fl4health/strategies/model_merge_strategy.py

```diff
@@ -0,0 +1,271 @@
+from logging import WARNING
```

Comment:
I'm probably missing something here, so please correct me if I'm wrong. Is there any reason this strategy can't inherit from `BasicFedAvg` and we just override methods like `initialize_parameters` and `configure_fit` to do what is needed for this strategy?

Reply:
Maybe the above is more trouble than its worth, but I think it could help eliminate some code duplication and also give the user flexibility to model merge with uniform or weighted FedAvg as well.

Reply:
Perhaps we can have a quick call tomorrow about this to hash it out but my thinking was we will also have to override aggregate_fit and aggregate_evaluate in addition to configure_fit and initialize_parameters because in this setting it seems to only make sense to do a uniform average. It feels a bit weird to average the model weights proportional to the amount of samples in the client test set. Perhaps it makes sense for the evaluation though. However, I also have to modify aggregate_evaluate to not compute a loss because we are not defining one. So at that point I have to override some of the core methods. And then on top of that, there is a few arguments that don't make sense for ModelMergeClient like initial_parameters and potentially weighted aggregation and weight eval losses depending on what we land on for supporting weighting for parameters and losses for model merging.

Reply:
I am open to extending the BasicFedAvg just wanted to add where I was coming from and see what you think! Lets discuss tomorrow :)

Reply:
Yeah I think that makes sense. I hadn't thought deeply about it. Was just thinking we might be able to reuse some code. Let's have a brief discussion today. However, your reasoning for not doing it above seems good to me.